lib/x509.js

```diff
@@ -3074,15 +3093,18 @@ pki.verifyCertificateChain = function(caStore, chain, verify) {
     var parent = null;
     var selfSigned = false;
 
-    // 1. check valid time
-    if(now < cert.validity.notBefore || now > cert.validity.notAfter) {
-      error = {
-        message: 'Certificate is not valid yet or has expired.',
-        error: pki.certificateError.certificate_expired,
-        notBefore: cert.validity.notBefore,
-        notAfter: cert.validity.notAfter,
-        now: now
-      };
+    if(validityCheckDate) {
+      // 1. check valid time
+      if(validityCheckDate < cert.validity.notBefore || 
+         validityCheckDate > cert.validity.notAfter) {
+        error = {
+          message: 'Certificate is not valid yet or has expired.',
+          error: pki.certificateError.certificate_expired,
+          notBefore: cert.validity.notBefore,
+          notAfter: cert.validity.notAfter,
+          now: validityCheckDate
```

Comment:
We should actually rename this from `now` to `validityCheckDate`. However, again, I'm not sure how strict you are about breaking the API. The options I see here are to
1. Rename `now` to `validityCheckDate`
2. Add a second field `validityCheckDate` and keep `now` for legacy reasons
3. Stick with `now` even though it is incorrect

What do you think?

Reply:
We would rather not change the API.  I think `now` is ok.  It's what you specified as "now".  Could leave a comment to revisit the name in the future if larger changes are happening.